addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -312,7 +312,7 @@ def _post_invoice_edi(self, invoices, test_mode=False):
     # -------------------------------------------------------------------------
 
     def _check_filename_is_fattura_pa(self, filename):
-        return re.search("([A-Z]{2}[A-Za-z0-9]{2,28}_[A-Za-z0-9]{0,5}.(xml.p7m|xml))", filename)
+        return re.search("([A-Z]{2}[A-Za-z0-9]{2,28}_[A-Za-z0-9]{0,5}.(?i:xml.p7m|xml))", filename)
```

Comment:
I don't think the ?i: will work here, it will make a non-capturing group which means you can't get it later in groups()
see https://regex101.com/r/3XdFVv/1

Reply:
Ah, guess he's right. Wasn't it tested?
```py
>>> re.search("hello\.(world)", "hello.WoRlD", re.IGNORECASE).groups()
('WoRlD',)
>>> re.search("hello\.(?i:world)", "hello.WoRlD").groups()
()
```

Reply:
Maybe it's time to write a small test for this :monocle_face:

Reply:
```suggestion
        return re.search("([A-Z]{2}[A-Za-z0-9]{2,28}_[A-Za-z0-9]{0,5}.((?i)xml.p7m|xml))", filename)
```
This solution is both capturing and case insensitive

Reply:
Why are we capturing the whole match ... at all?

Reply:
:man_shrugging: I'm not sure if it was needed somewhere else, otherwise the outer parentheses can indeed be removed as long as we don't forget to update the groups()[1]
```suggestion
        return re.search("[A-Z]{2}[A-Za-z0-9]{2,28}_[A-Za-z0-9]{0,5}.((?i)xml.p7m|xml)", filename)
```

Reply:
I need to correct myself, (?i) is going to impact the whole string in python which is not what we want.
A solution that seems ok is to... put the non-capturing group in a capturing group :sweat_smile: 
```suggestion
        return re.search("([A-Z]{2}[A-Za-z0-9]{2,28}_[A-Za-z0-9]{0,5}.((?i:xml.p7m|xml)))", filename)
```